index.js

```diff
@@ -0,0 +1,205 @@
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+      jsx: true,
+    },
+    sourceType: 'module',
+    project: './tsconfig.json',
+  },
+  extends: [
+    'plugin:@typescript-eslint/recommended',
+    'prettier/@typescript-eslint',
+    'plugin:prettier/recommended',
+  ],
+  rules: {
+    '@typescript-eslint/array-type': ['error', { default: 'generic', readonly: 'generic' }],
+    '@typescript-eslint/ban-ts-comment': 'error',
+    '@typescript-eslint/consistent-type-assertions': 'error',
+    '@typescript-eslint/explicit-member-accessibility': 'error',
+    '@typescript-eslint/indent': 'off',
+    '@typescript-eslint/naming-convention': [
+      'error',
+      {
+        selector: 'default',
+        format: ['camelCase'],
+        leadingUnderscore: 'forbid',
+        trailingUnderscore: 'forbid',
+      },
+      {
+        selector: 'typeLike',
+        format: ['PascalCase'],
+      },
+      {
+        selector: 'variable',
+        format: ['camelCase', 'UPPER_CASE'],
+      },
+      {
+        selector: 'enumMember',
+        format: ['PascalCase'],
+      },
+    ],
+    '@typescript-eslint/no-empty-function': 'error',
+    '@typescript-eslint/no-explicit-any': 'error',
+    '@typescript-eslint/no-inferrable-types': 'off',
+    '@typescript-eslint/no-unused-vars': [
+      'error',
+      { vars: 'all', args: 'after-used', ignoreRestSiblings: true },
+    ],
+    '@typescript-eslint/no-use-before-define': 'error',
+    '@typescript-eslint/prefer-interface': 'off',
+    '@typescript-eslint/prefer-readonly': 'error',
+    camelcase: 'off',
+  },
+};
+
+const reactSettings = {
+  extends: ['plugin:react/recommended'],
+  rules: {
+    'react/button-has-type': 'error',
+    'react/display-name': 'off',
+    'react/jsx-boolean-value': 'error',
+    'react/jsx-curly-brace-presence': ['error', { props: 'never', children: 'never' }],
+    'react/jsx-filename-extension': ['error', { extensions: ['.jsx', '.tsx'] }],
+    'react/jsx-wrap-multilines': 'off',
+    'react/no-array-index-key': 'error',
+    'react/no-this-in-sfc': 'error',
+    'react/no-unused-prop-types': 'error',
+    'react/no-unused-state': 'error',
+    'react/void-dom-elements-no-children': 'error',
+  },
+};
+
+module.exports = {
+  extends: ['airbnb', 'plugin:prettier/recommended'],
+  plugins: ['import', 'unicorn'],
+  settings: {
+    react: {
+      version: 'detect',
+    },
+  },
+  env: {
+    browser: true,
+    es6: true,
+  },
+  globals: {
+    require: true,
+    process: true,
+  },
+  rules: {
+    'default-case': 'error',
+    'import/extensions': [
+      'error',
+      'always',
+      { js: 'never', jsx: 'never', ts: 'never', tsx: 'never', hbs: 'never', vue: 'never' },
+    ],
+    'import/no-unresolved': 'error',
+    'import/order': ['error', { groups: ['external', 'builtin', ['sibling', 'parent']] }],
+    'import/prefer-default-export': 'off',
+    'lines-between-class-members': 'off',
+    'max-lines': 'error',
+    'no-console': 'error',
```

Comment:
Do we also want to error on usages like `.error` `.info` whats the reasoning behind disallowing these as well ?

Reply:
`console` statements are for debugging purposes, so they should not end up into a production build. If you have a very good reason to have a `console` in your code, than you can disable the rule for that line. But you should be aware that it is an exception.

Reply:
Yeah but most of us `log` function. Writing `.warn`, `.info`, `.error` imho are thought over logs. You could of course misuse this and write everywhere `info` instead of `log`. That's what I meant with disallowing only `.log` it's possible with that rule.

Reply:
While I agree with @jesse-mm that info/warn/error are used less for debugging, and more for information purposes... If we want to be strict about it, we should force those logs into a Logging Library where they can be turned on conditionally per "namespace" or log level. So when you go to production, the console is clean.